src/async-wrap.cc

```diff
@@ -181,6 +183,33 @@ static void Initialize(Local<Object> target,
 }
 
 
+void AsyncWrap::DestroyIdsCb(uv_idle_t* handle) {
+  Environment* env = Environment::from_destroy_ids_idle_handle(handle);
```

Comment:
You're right. Something that subtle could easily lead to future pain. Will add.

On the side, I'm surprised that both `Context::Scope` and `TryCatch` don't leak a handle. Think the `Context::Scope` would still work properly without the `HandleScope`?